File: BGVClient/src/vga.jl

encode(a::Multivector{3,1}) = Dict(
	"Arrow Vector" => [Dict("Vector"=>Vector(a.comps))]
)

encode(a::Multivector{3,1}) = (rig="Arrow Vector", )

encode(a::Multivector{3,2}) = Dict(
	"Circle 2-blade" => [Dict(
		"Normal"=>Vector(rdual(a).comps),
		"Radius"=>sqrt(sqrt(abs(a⊙a))),
	)]
)

File: python/geomviz/models/vga.py
# 3D vector/vanilla geometric algebra

from math import sqrt

def encode(mv):
	if mv.grades() == {1}:
		return {
		'rig_name': "Arrow Vector",
		'rig_parameters': {
			"Vector": mv.value[1:4].tolist()
		}
	}
	elif mv.grades() == {2}:
		return {
		'rig_name': "Circle 2-blade",
		'rig_parameters': {
			"Normal": mv.value[5:8].tolist(),
			"Radius": sqrt(abs(mv))
		}
	}
	else:
		raise NotImplementedError
